streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -52,21 +52,27 @@
     private JoinWindows(String name, long before, long after) {
         super(name);
 
+        if (before < 0) {
+            throw new IllegalArgumentException("window size must be > 0 (you provided before as " + before + ")");
+        }
+        if (after < 0) {
+            throw new IllegalArgumentException("window size must be > 0 (you provided after as " + after + ")");
+        }
+        if (before == 0 && after == 0) {
```

Comment:
@miguno @dguy 
A window must not be symmetric: ie, having zero for before or after and a positive value for the other is ok: `[0,5]`, `[5,0]` both ok -- `[0,0]` and negative values are prohibited. Makes sense? Maybe the cases can be collapsed into a single condition -- to me, having multiple ìf` was more natural.

Does this make sense?